php/Blockonomics.php

```diff
@@ -217,85 +323,37 @@ private function set_headers($api_key)
+        if ($bch_enabled  == '1'){
+            return 'bch';
         }else{
-            $response_callback = '';
-            $response_address = '';
-        }
-        $callback_secret = get_option('blockonomics_callback_secret');
-        $api_url = WC()->api_request_url('WC_Gateway_Blockonomics');
-        $callback_url = add_query_arg('secret', $callback_secret, $api_url);
-        // Remove http:// or https:// from urls
-        $api_url_without_schema = preg_replace('/https?:\/\//', '', $api_url);
-        $callback_url_without_schema = preg_replace('/https?:\/\//', '', $callback_url);
-        $response_callback_without_schema = preg_replace('/https?:\/\//', '', $response_callback);
-        //TODO: Check This: WE should actually check code for timeout
-        if (!wp_remote_retrieve_response_code($response)) {
-            $error_str = __('Your server is blocking outgoing HTTPS calls', 'blockonomics-bitcoin-payments');
-        }
-        elseif (wp_remote_retrieve_response_code($response)==401)
-            $error_str = __('API Key is incorrect', 'blockonomics-bitcoin-payments');
-        elseif (wp_remote_retrieve_response_code($response)!=200)  
-            $error_str = $response->data;
-        elseif (!isset($response_body) || count($response_body) == 0)
-        {
-            $error_str = __('You have not entered an xPub', 'blockonomics-bitcoin-payments');
+            return 'btc';
         }
-        elseif (count($response_body) == 1)
-        {
-            if(!$response_callback || $response_callback == null)
-            {
-              //No callback URL set, set one 
-              $this->update_callback($callback_url, $response_address);   
-            }
-            elseif($response_callback_without_schema != $callback_url_without_schema)
-            {
-              $base_url = get_bloginfo('wpurl');
-              $base_url = preg_replace('/https?:\/\//', '', $base_url);
-              // Check if only secret differs
-              if(strpos($response_callback, $base_url) !== false)
-              {
-                //Looks like the user regenrated callback by mistake
-                //Just force Update_callback on server
-                $this->update_callback($callback_url, $response_address);  
-              }
-              else
-              {
-                $error_str = __("You have an existing callback URL. Refer instructions on integrating multiple websites", 'blockonomics-bitcoin-payments');
-              }
-            }
-        }
-        else 
-        {
-            $error_str = __("You have an existing callback URL. Refer instructions on integrating multiple websites", 'blockonomics-bitcoin-payments');
-            // Check if callback url is set
-            foreach ($response_body as $res_obj)
-             if(preg_replace('/https?:\/\//', '', $res_obj->callback) == $callback_url_without_schema)
-                $error_str = "";
-        }  
```

Comment:
The final else statement has been removed during the refactoring, however, this code is required to check for an existing callback URL, if the user has multiple xpub addresses `count($response_body) > 1`. This code should be reinstated into check_get_callbacks_response_body

Reply:
Thanks for letting me know. I reinstated the else statement into check_get_callbacks_response_body.